pkg/cloud/aws/actuators/scope.go

```diff
@@ -59,17 +65,25 @@ func NewScope(params ScopeParams) (*Scope, error) {
 		return nil, errors.Errorf("failed to load cluster provider status: %v", err)
 	}
 
-	session, err := session.NewSession(aws.NewConfig().WithRegion(clusterConfig.Region))
-	if err != nil {
-		return nil, errors.Errorf("failed to create aws session: %v", err)
+	sessionLock.Lock()
+	defer sessionLock.Unlock()
+
+	var localSession *session.Session
+	if cachedSession, ok := sessionCache[clusterConfig.Region]; !ok {
+		localSession, err = session.NewSession(aws.NewConfig().WithRegion(clusterConfig.Region))
+		if err != nil {
+			return nil, errors.Errorf("failed to create aws session: %v", err)
```

Comment:
When will we remove entries from the cache?

Reply:
AFAIK that should not be necessary

Reply:
If we do have some type of ttl, then we can avoid having to scale/redeploy the controller StatefulSet to pick up a change to the credentials (if mounted from a Secret).

Reply:
good point! no harm in having a ttl

Reply:
Do we need this for v1 of the session cache?

Reply:
No, but we should create a tracking issue.